salt/crypt.py

```diff
@@ -1291,6 +1292,8 @@ def sign_in(self, timeout=60, safe=True, tries=1, channel=None):
         if not channel:
             close_channel = True
             channel = salt.transport.client.ReqChannel.factory(self.opts, crypt='clear')
```

Comment:
this could also benefit from a channel RAII object, because what you mainly want is if you create it here, to close it, otherwise let who created it to close it. That's perfect for RAII